drivers/lsm6ds3/lsm6ds3.c

```diff
@@ -0,0 +1,296 @@
+                         LSM6DS3_REG_CTRL3_C, LSM6DS3_CTRL3_C_BDU);
+    DEBUG("LSM6DS3: retrieve gyroscope values ... ");
+    res += i2c_read_reg(dev->i2c, dev->gyr_address,
+                        LSM6DS3_REG_OUT_X_L_G, &tmp);
+    data->x_axis = tmp;
+    res += i2c_read_reg(dev->i2c, dev->gyr_address,
+                        LSM6DS3_REG_OUT_X_H_G, &tmp);
+    data->x_axis = tmp<<8;
+    res += i2c_read_reg(dev->i2c, dev->gyr_address,
+                        LSM6DS3_REG_OUT_Y_L_G, &tmp);
+    data->y_axis = tmp;
+    res += i2c_read_reg(dev->i2c, dev->gyr_address,
+                        LSM6DS3_REG_OUT_Y_H_G, &tmp);
+    data->y_axis = tmp<<8;
+    res += i2c_read_reg(dev->i2c, dev->gyr_address,
+                        LSM6DS3_REG_OUT_Z_L_G, &tmp);
+    data->z_axis = tmp;
+    res += i2c_read_reg(dev->i2c, dev->gyr_address,
+                        LSM6DS3_REG_OUT_Z_H_G, &tmp);
+    data->z_axis = tmp<<8;
+    res += i2c_write_reg(dev->i2c, dev->gyr_address,
+                         LSM6DS3_REG_CTRL3_C, ~(1<<7));
+	  i2c_release(dev->i2c);
+
+    data->x_axis = data->x_axis>>4;
+    data->y_axis = data->y_axis>>4;
+    data->z_axis = data->z_axis>>4;
+
+	  if (res != 8)
+	  {
+		    DEBUG("[ FAIL ]\n");
+		    return -1;
+	  }
+
+	  DEBUG("[ OK ]\n");
+  	return 0;
+}
+
+int lsm6ds3_read_temp(lsm6ds3_t *dev, int16_t *value)
+{
+	  int res = 0;
+    char tmp;
+
+	  i2c_acquire(dev->i2c);
+    i2c_read_reg(dev->12c, dev-acc_address, LSM6DS3_STATUS_REG, &tmp);
+    DEBUG("LSM6DS3: status ... %x\n", tmp);
+    res += i2c_write_reg(dev->i2c, dev->acc_address,
+                         LSM6DS3_REG_CTRL3_C, LSM6DS3_CTRL3_C_BDU);
+    DEBUG("LSM6DS3: retrieve temperature data ... ");
+    res += i2c_read_reg(dev->i2c, dev->acc_address,
+                        LSM6DS3_OUT_TEMP_L, &tmp);
+    *value += tmp;
+    res += i2c_read_reg(dev->i2c, dev->acc_address,
+                        LSM6DS3_OUT_TEMP_H, &tmp);
+    *value += tmp<<8;
+    res += i2c_write_reg(dev->i2c, dev->acc_address,
+                         LSM6DS3_REG_CTRL3_C, ~(1<<7));
+	  i2c_release(dev->i2c);
+
+	  if (res != 2)
+	  {
+		    DEBUG("[ FAIL ]\n");
+		    return -1;
+	  }
+
+	  *value = (((*value) >> 8) & 0ff | (*value << 8);
+	  DEBUG("raw temperature: %i\n", *value);
+    return 0;
+}
+
+int lsm6ds3_disable(lsm6ds3_t *dev)
+{
+	  int res = 0;
+    char tmp = (LSM6DS3_CTRL4_C_SLEEP_G
+               /*| (~LSM6DS3_CTRL4_C_TEMP_EN)*/);
+
+	  i2c_acquire(dev->i2c);
+    DEBUG("LSM6DS3: disable device (%i) ... ", *dev);
+    res += i2c_write_reg(dev->i2c, dev->acc_address,
+                         LSM6DS3_REG_CTRL1_XL, LSM6DS3_CTRL1_XL_ODR_POWEROFF);
+    res += i2c_write_reg(dev->i2c, dev->gyr_address,
+                         LSM6DS3_REG_CTRL2_G, LSM6DS3_CTRL2_G_ODR_POWEROFF);
+	  res += i2c_write_reg(dev->i2c, dev->acc_address,
+                         LSM6DS3_REG_CTRL4_C, tmp);
+    i2c_release(dev->i2c);
+
+	  if (res != 2)
+	  {
+		    DEBUG("[ FAIL ]\n");
```

Comment:
BTW: If you like to write many DEBUG messages, could you give some details inside? So one directly knows where it comes from